pkg/controllers/management/aks/aks_cluster_handler_test.go

```diff
@@ -0,0 +1,297 @@
+	- default phase
+	- create phase
+	- active phase
+	- update node pool phase
+ */
+func Test_onClusterChange_ClusterIsNil(t *testing.T) {
+	cluster, _ := mockOperatorController.onClusterChange("", nil)
+	if cluster != nil {
+		t.Errorf("cluster should have returned nil")
+	}
+}
+
+func Test_onClusterChange_AKSConfigIsNil(t *testing.T) {
+	mockCluster := &v3.Cluster{
+		Spec: v3.ClusterSpec{
+			AKSConfig: nil,
+		},
+	}
+
+	cluster, _ := mockOperatorController.onClusterChange("", mockCluster)
+	if !reflect.DeepEqual(cluster, mockCluster) {
+		t.Errorf("cluster should have returned with no update")
+	}
+}
+
+func Test_onClusterChange_Default(t *testing.T) {
+
+	// setup
+	// create an instance of the operator controller with mock data to simulate the onChangeCluster function reacting
+	// to a real cluster!
+	mockOperatorController = getMockAksOperatorController("default")
+
+	mockCluster, err := getMockV3Cluster(MockDefaultClusterFilename)
+	if err != nil {
+		t.Errorf("error getting mock v3 cluster: %s", err)
+	}
+
+	// run test
+	cluster, err := mockOperatorController.onClusterChange("", &mockCluster)
+
+	// validate results
+	if err != nil {
+		t.Errorf("error running onClusterChange: %s", err)
+	}
+	if cluster.Status.Conditions[0].Status != mockConditionUnknown.Status {
+		t.Errorf("provisioned status should be Unknown and cluster returned successfully")
+	}
+}
+
+func Test_onClusterChange_Create(t *testing.T) {
+	mockOperatorController = getMockAksOperatorController("create")
+	mockCluster, err := getMockV3Cluster(MockCreateClusterFilename)
+	if err != nil {
+		t.Errorf("error getting mock v3 cluster: %s", err)
+	}
+
+	cluster, err := mockOperatorController.onClusterChange("", &mockCluster)
+
+	if err != nil {
+		t.Errorf("error running onClusterChange: %s", err)
+	}
+	if cluster.Status.Conditions[0].Status != mockConditionUnknown.Status {
+		t.Errorf("provisioned status should be Unknown and cluster returned successfully")
+	}
+}
+
+func Test_onClusterChange_Active(t *testing.T) {
+	mockOperatorController = getMockAksOperatorController("active")
+	mockCluster, err := getMockV3Cluster(MockActiveClusterFilename)
+	if err != nil {
+		t.Errorf("error getting mock v3 cluster: %s", err)
+	}
+
+	cluster, err := mockOperatorController.onClusterChange("", &mockCluster)
+
+	if err != nil {
+		t.Errorf("error running onClusterChange: %s", err)
+	}
+	if cluster.Status.Conditions[0].Status != mockConditionTrue.Status || cluster.Status.Conditions[1].Status != mockConditionTrue.Status {
+		t.Errorf("provisioned and updated status should be True and cluster returned successfully")
+	}
+}
+
+func Test_onClusterChange_UpdateNodePool(t *testing.T) {
+	mockOperatorController = getMockAksOperatorController("update")
+	mockCluster, err := getMockV3Cluster(MockUpdateClusterFilename)
+	if err != nil {
+		t.Errorf("error getting mock v3 cluster: %s", err)
+	}
+
+	cluster, err := mockOperatorController.onClusterChange("", &mockCluster)
```

Comment:
The cluster read from the file `test/onclusterchange_update.yaml` does not contain the field `status.driver`, so the function `mockOperatorController.onClusterChange` returns the cluster without touching the cluster's node pool or status at the following lines:
https://github.com/rancher/rancher/blob/68215bcdde090854ff28b43b01d1aa0b69611920/pkg/controllers/management/aks/aks_cluster_handler.go#L92 

That means the function does not validte "updateNodePool", right? 
Or am I misunderstanding anything?